examples_tests/48.ArithmeticUnitTest/main.cpp

```diff
@@ -0,0 +1,230 @@
+		/*if (val != *i)
+			goto validationEndFalse;*/
+		invocationIndex++;
+	}
+	downloadStagingArea->multi_free(1u, &address, &BUFFER_SIZE, nullptr);
+	return true;
+
+validationEndFalse:
+	downloadStagingArea->multi_free(1u, &address, &BUFFER_SIZE, nullptr);
+	return false;
+
+}
+int main()
+{
+	// create device with full flexibility over creation parameters
+	// you can add more parameters if desired, check irr::SIrrlichtCreationParameters
+	irr::SIrrlichtCreationParameters params;
+	params.Bits = 24; //may have to set to 32bit for some platforms
+	params.ZBufferBits = 24; //we'd like 32bit here
+	params.DriverType = video::EDT_OPENGL; //! Only Well functioning driver, software renderer left for sake of 2D image drawing
+	params.WindowSize = dimension2d<uint32_t>(1280, 720);
+	params.Fullscreen = false;
+	params.Vsync = true; //! If supported by target platform
+	params.Doublebuffer = true;
+	params.Stencilbuffer = false; //! This will not even be a choice soon
+	params.StreamingDownloadBufferSize = BUFFER_SIZE * 2;
+	auto device = createDeviceEx(params);
+
+	if (!device)
+		return 1; // could not create selected driver.
+
+	video::IVideoDriver* driver = device->getVideoDriver();
+	io::IFileSystem* filesystem = device->getFileSystem();
+	asset::IAssetManager* am = device->getAssetManager();
+
+	uint32_t* inputData = new uint32_t[BUFFER_SIZE / sizeof(uint32_t)];
+
+	auto inputDataBuffer = make_smart_refctd_ptr<ICPUBuffer>(BUFFER_SIZE);
+	{
+		uint32_t* ptr = static_cast<uint32_t*>(inputDataBuffer->getPointer());
+		for (size_t i = 0; i < BUFFER_SIZE / sizeof(uint32_t); i++)
+		{
+			auto memAddr = ptr + i;
+			uint32_t randomValue = std::rand();
+			*memAddr = randomValue;
+			inputData[i] = randomValue;
+		}
+	}
+	core::smart_refctd_ptr<IGPUBuffer> gpuinputDataBuffer = core::smart_refctd_ptr<IGPUBuffer>(driver->getGPUObjectsFromAssets(&inputDataBuffer, &inputDataBuffer + 1)->front()->getBuffer());
+	//create 7 buffers.
+	core::smart_refctd_ptr<IGPUBuffer> buffers[7];
+	for (size_t i = 0; i < 7; i++)
+	{
+		buffers[i] = driver->createDeviceLocalGPUBufferOnDedMem(BUFFER_SIZE);
+	}
+
+	IGPUDescriptorSetLayout::SBinding binding[8] = {
+		{0u,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},	//input with randomized numbers
+		{TO_AND,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_XOR,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_OR,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_ADD,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_MUL,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_MIN,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_MAX,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+	};
+	auto gpuDSLayout = driver->createGPUDescriptorSetLayout(binding, binding + 8);
+	constexpr uint32_t pushconstantSize = 1;
```

Comment:
IIRC push constants size are given in bytes and sizes+offsets need to be aligned to 4 bytes